src/main/java/com/commercetools/sync/commons/utils/ClientConfigurationUtils.java

```diff
@@ -46,6 +56,36 @@ public static BlockingSphereClient createClient(@Nonnull final SphereClientConfi
     }
 
     /**
+     * Return a {@link SphereClient} instance which will retry its request after it receives a server error from CTP.
+     * A given {@code sphereClient} is wrapped by {@link RetrySphereClientDecorator} with proper rules initialized, and
+     * then returned.
+     *
+     * @param sphereClient a delegate that will execute requests
+     * @return {@link SphereClient} that will retry its requests on 5xx errors
+     */
+    @Nonnull
+    public static SphereClient getRetryOn5xxClient(@Nonnull final SphereClient sphereClient) {
+        return RetrySphereClientDecorator.of(sphereClient, singletonList(RetryRule.of(
+            ofMatchingErrors(ServerErrorException.class),
+            ofScheduledRetry(RETRY_ON_5XX_ATTEMPTS_LIMIT, ClientConfigurationUtils::retryOn5xxDurationCalculator))));
+    }
+
+    /**
+     * Given a {@link RetryContext} it returns duration for a current attempt. The duration is calculated from a
+     * following equation: <pre>duration = 2 ^ (((retryAttemptNumber - 1) / 3) + 1) [seconds]</pre>
```

Comment:
About the strategy: it is only my proposition. In fact I have no clue what retry strategy is wanted here. I have expressed my concerns some time ago in a issue https://github.com/commercetools/commercetools-sync-java/issues/31#issuecomment-306434605

This particular strategy seemed reasonable to me: first it performs few attempts with a little duration and then the time is increased. There is 27 attempts so it sum to almost 2 hours of waiting (e.g. when the server is down). 

Please share your opinion about what is expected here @heshamMassoud